sos/policies/__init__.py

```diff
@@ -49,6 +51,107 @@ def load(cache={}, sysroot=None):
     return cache['policy']
 
 
+class ContainerRuntime(object):
+    """Encapsulates a container runtime that provides the ability to plugins to
+    check runtime status, check for the presence of specific containers, and
+    to format commands to run in those containers
+    """
+
+    runtime_name = 'Undefined'
+    containers = []
+    images = []
+    volumes = []
+    binary = ''
+
+    def __init__(self, policy=None):
+        self.policy = policy
+        self.run_cmd = "%s exec " % self.binary
+
+        self.active = self._check_is_active()
+        self.containers = self.get_containers()
+        self.images = self.get_images()
+        self.volumes = self.get_volumes()
+
+    def _check_is_active(self):
+        """Check to see if the container runtime is both present AND active.
+
+        Active in this sense means that the runtime can be used to glean
+        information about the runtime itself and containers that are running.
+        """
+        return is_executable(self.binary)
+
+    def get_containers(self, get_all=False):
+        """Get a list of containers present on the system.
+
+        If `get_all` is `True`, also include non-running containers
+        """
+        containers = []
+        _cmd = "%s ps %s" % (self.binary, '-a' if get_all else '')
+        if self.active:
+            out = sos_get_command_output(_cmd)
+            if out['status'] == 0:
+                for ent in out['output'].splitlines()[1:]:
+                    ent = ent.split()
+                    containers.append((ent[0], ent[-1]))
+        return containers
+
+    def get_container_by_name(self, name):
+        """Get the container ID for the container matching the provided
+        name
+        """
+        if not self.active or name is None:
+            return None
+        for c in self.containers:
+            if re.match(name, c[1]):
```

Comment:
Will be the reg.exp. utilized or can't be just string comparison sufficient? Just for performance reasons..